FE/src/components/Add/DependencyFormInput.tsx

```diff
@@ -0,0 +1,127 @@
+import { useState } from "react";
+import {TextField, Box, Chip, Paper, List, ListItem, ListItemButton, ListItemText, IconButton} from "@mui/material";
+import CancelIcon from "@mui/icons-material/Cancel";
+import ClearIcon from "@mui/icons-material/Clear";
+
+export default function DependencyFormInput({label, selectedItems, setSelectedItems, availableItems}: {label: string; selectedItems: string[]; setSelectedItems: (items: string[]) => void; availableItems: string[]}) {
```

Comment:
So you're passing a reactive variable to another component for modification? Why not instantiate a store for said reactive variable and retrieve it from that store? (That's how it worked in Vue at least, but React's gotta have stores as well, right?)

More importantly, define a type for the props. Keeps the code short and more readable.
```suggestion
type DependencyFormProps = {
         label: string;
         selectedItems: string[];
         ...
}
export default function DependencyFormInput({label, selectedItems, setSelectedItems, availableItems}: DependencyFormProps) {
```

Reply:
Which variable are we talking about?

Reply:
Props is fine, I just think the opposite, I can change it

Reply:
selectedItems

Reply:
Why make a store for it?

Reply:
From what I remember, the selected items are just the dependencies for the tasks. If we want to create a task, we need a simple empty list. If we update, we get the dependencies into it. How would a store* help this?

Reply:
I don't know what you mean. I thought it's best practice to store reactive variables that pertain to multiple components in stores like Zustand (React) or Pinia (Vue). I think it would make the code cleaner plus we'll probably need Zustand in the future for global variables.

I guess you can leave it as is for now 'cause I'm not a React scholar (yet).

Reply:
React has the context for globals. We could use it for keeping the Tasks if we don't want to always call fetch all or things like that. The selected items would then be the dependencies from the Task (for update). Thing is, selected items also show un the UI, so storing them in a useState is a must (from what I know). You can check the figma design in the add part.

Reply:
Okay I think I do get your initial point now, and we will probably have either Zustand or Context in the future, I had to work with 'lib.ts' file that had a in memory repo for this sprint. Then the only difference would be indeed the retrieval of stuff from there.